src/scripts/quests/questTypes/DefeatGymQuest.ts

```diff
@@ -13,13 +13,17 @@ class DefeatGymQuest extends Quest implements QuestInterface {
         this.focus = App.game.statistics.gymsDefeated[GameConstants.getGymIndex(this.gymTown)];
     }
 
+    // Only add Defeat Gym Quest if the player has defeated the first gym (Brock).
+    public static canComplete() {
+        return App.game.badgeCase.hasBadge(BadgeEnums.Boulder);
+    }
+
     public static generateData(): any[] {
         const amount = SeededRand.intBetween(5, 20);
         const region = SeededRand.intBetween(0, player.highestRegion());
```

Comment:
Region would need to be restricted, if you roll your highest (non-kanto) region but do not have any gym clears in that region, `calcReward` fails because `possibleGyms` is an empty array so `gymTown` is `undefined`

![image](https://github.com/pokeclicker/pokeclicker/assets/16630222/173434b9-9a5f-4472-adf7-135abd57a912)

Reply:
Hm... So I guess I should have kept that part of the logic. I see now why it was there. Oops.